tests/conftest.py

```diff
@@ -0,0 +1,277 @@
+import pytest
+import vcr
+import json
+from blaseball_mike.models import Player, Game, Team, Fight,\
+    Election, ElectionResult, DecreeResult, BlessingResult, TidingResult,\
+    SeasonStatsheet, GameStatsheet, TeamStatsheet, PlayerStatsheet,\
+    Playoff, PlayoffRound, PlayoffMatchup,\
+    League, Subleague, Division, Tiebreaker,\
+    Idol, Tribute, Standings, GlobalEvent, Season, SimulationData
+from .helpers import TEST_DATA_DIR
+
+CASSETTE_DIR = f'{TEST_DATA_DIR}/cassettes'
+
+
+@pytest.fixture(scope="module")
+def vcr_config():
+    return {
+        "cassette_library_dir": CASSETTE_DIR,
+        }
+
+
+@pytest.fixture(scope="module")
+@vcr.use_cassette(f'{CASSETTE_DIR}/Fixture.games.yaml')
+def games():
```

Comment:
Do you want to have the static ones (like crowvertime) load the `dict` from a json or do you want it defined in `conftest.py`? I worry having it hard-coded will just bloat the config file.

The reason I used the singular JSON for each was to simplify adding new test cases. You only have to append it to one file rather than adding a JSON file and adding a fixture and adding that fixture to the group fixture.

Reply:
imo a bloated conftest is fine (that's a problem solved by the interpreter, so let's not reinvent the wheel), so yeah, static ones can be moved in there.

if the json file is removed, adding new cases is just two touch points (new fixture, append to group fixture), though as mentioned in other comments, it would be best if we could take advantage of parametrization at the test level, which solves the group fixture touch point. 

My general reasoning for preferring a bunch of small fixtures is that if a new test case arises, that really should only add one test for that case, not n+1 tests for all the existing prior cases. Those prior cases should, in theory, already be covered. 

Like for example, let's say there's a bug found in calculating payouts on sun2 games--therefore we should only test payouts on a sun2 game when making the fix. We don't need to test initialization as there's no special case, we don't need to test player mods as weather loading and player loading are independent, etc.

Reply:
New update has split the stuff out. The problem is you cant call a fixture inside a fixture, so it's currently setup that the *group* is a fixture and the rest are just standard functions. This isn't ideal, since it means you cant explicitly call specific fixtures for one-off tests. I also dont think the group can be parameterized if it's a fixture, unless we do the `pytest_generate_tests` stuff again...

Reply:
wait you can't call a fixture inside a fixture? i feel like i have done this before....

what i was hoping for was that the for loop could be taken out of the individual test cases and just rely on the parametrization scaffolding, so now we're kind of in a worst-of-both-worlds situation.

some cursory poking around, I found a thing that I've never used. I wonder if's any good? https://pypi.org/project/pytest-lazy-fixture/

Reply:
I *think* I solved it. So if we define the singular test functions as fixtures, then perform parametrization *on the group fixture*, we can do this:

```python
@pytest.fixture(scope="module", params=['game_s7d95', 'game_s2d99_chronicler', 'game_s9d1_crovertime', 'game_s2d10_none_awaypitcher', 'game_coffee_cup_d11'])
def game(request):
    return request.getfixturevalue(request.param)
```

So any test function that calls `game` as an argument will be automatically parametrized with the tests in the `params` list.

Reply:
ok yeah, i think this works pretty well, even though it looks like a mess. as long as the mess is contained to conftest, I think that's a win.

Reply:
OK, I've done enough of them that the test dont error. I'll go back and finish up the last of them after dinner.

Is there any interest in moving these out of conftest.py? The tests are grouped by class so we could just put the fixtures in with their respective tests.

Reply:
yeah separating them out is probably fine to keep it svelt